src/main/java/com/exactpro/th2/conn/ConnectivityBatch.java

```diff
@@ -81,21 +81,36 @@ private static void checkMessages(List<ConnectivityMessage> iMessages, String se
         if (!iMessages.stream()
                 .allMatch(iMessage -> Objects.equals(sessionAlias, iMessage.getSessionAlias())
                         && direction == iMessage.getDirection())) {
-            throw new IllegalArgumentException("List " + iMessages + " has elemnts with incorrect metadata, expected session alias '"+ sessionAlias +"' direction '" + direction + '\'');
+            throw new IllegalArgumentException("List " + iMessages + " has elements with incorrect metadata, expected session alias '"+ sessionAlias +"' direction '" + direction + '\'');
         }
 
-        if (!IntStream.range(0, iMessages.size() - 1)
-                .allMatch(index -> iMessages.get(index).getSequence() + 1 == iMessages.get(index + 1).getSequence())) {
-            if (LOGGER.isErrorEnabled()) {
-                LOGGER.error("List {} hasn't elements with incremental sequence with one step", iMessages.stream()
-                                            .map(ConnectivityMessage::getSequence)
-                                            .collect(Collectors.toList()));
+        boolean sequencesUnordered = false;
```

Comment:
We can check the value of missedSequences.isEmpty method instead

Reply:
In case we have descending (3, 2, 1) sequences or repeating (1, 1, 1) `missedSequences` will be empty.
Old approach shows error in this case.